src/system_interfaces/engine.ts

```diff
@@ -78,3 +96,55 @@ async function tauriRenderSceneWled(
   });
   return fromBinary(WledRenderTargetSchema, new Uint8Array(renderTargetBin));
 }
+
+/**
+ * Initialize Tauri render event listeners.
+ * Call this once when the app starts in Tauri mode.
+ * Returns an unregister function to clean up listeners.
+ */
+export async function initRenderListeners(): Promise<(() => void) | null> {
+  if (!isTauri) {
+    return null;
+  }
+
+  const unlisteners: UnlistenFn[] = [];
+
+  // Listen for DMX render events from Tauri backend
+  const unregisterDmxListener = await listen<DmxRenderEvent>(
+    'dmx-render',
+    (event) => {
+      const payload = event.payload;
+      const outputId = BigInt(payload.output_id);
+      const data = new Uint8Array(payload.data);
+
+      // Trigger subscriptions in renderRouter
+      triggerDmxSubscriptions(outputId, data);
+    },
+  );
+  unlisteners.push(unregisterDmxListener);
+
+  // Listen for WLED render events from Tauri backend
+  const unregisterWledListener = await listen<WledRenderEvent>(
+    'wled-render',
+    (event) => {
+      const payload = event.payload;
+      const outputId = BigInt(payload.output_id);
+      const data = fromBinary(
+        WledRenderTargetSchema,
+        new Uint8Array(payload.data),
+      );
+
+      // Trigger subscriptions in renderRouter
+      triggerWledSubscriptions(outputId, data);
+    },
+  );
+  unlisteners.push(unregisterWledListener);
+
+  console.log('Tauri render event listeners initialized');
+
+  // Return cleanup function
+  return () => {
+    unlisteners.forEach((unregister) => unregister());
+    console.log('Tauri render event listeners cleaned up');
+  };
```

Comment:
This return function is never called. Lets not worry about tracking or unregistered the listeners, they will exist for the lifetime of the application.